### PR TITLE
Refactor `backend.js` #3 - legibility

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING REVIEW

### What does this PR accomplish?
- This is the third step in https://github.com/metabase/metabase/issues/24023
- It extracts flags and configs for better legibility
- It doesn't change the logic
- It annotates some obscured behavior that was easy to miss due to `MB_USER_DEFAULTS` environment variable not being documented anywhere

### CODE REVIEW

File: frontend/test/__runner__/backend.js [hunk 1]
```diff
@@ -41,39 +41,58 @@ const BackendResource = createSharedResource("BackendResource", {
...
+        MB_DB_TYPE: "h2",
+        MB_DB_FILE: server.dbFile,
+        MB_JETTY_HOST: "0.0.0.0",
+        MB_JETTY_PORT: server.port,
+        MB_ENABLE_TEST_ENDPOINTS: "true",
+        MB_PREMIUM_EMBEDDING_TOKEN:
+          (process.env["MB_EDITION"] === "ee" &&
+            process.env["ENTERPRISE_TOKEN"]) ||
+          undefined,
+      };
```
Comment (kulyk) [inline frontend/test/__runner__/backend.js:62]: Not sure all of these settings are related to databases (as they're in `databaseConfig`). Meaning:

- `MB_JETTY_HOST`
- `MB_JETTY_PORT`
- `MB_ENABLE_TEST_ENDPOINTS`
- `MB_PREMIUM_EMBEDDING_TOKEN`

File: frontend/test/__runner__/backend.js [hunk 2]
```diff
@@ -8,15 +8,15 @@ const { spawn } = require("child_process");
 const fetch = require("isomorphic-fetch");
 
 let testDbId = 0;
-const getDbFile = () =>
+const generateTempDbFile = () =>
   path.join(os.tmpdir(), `metabase-test-${process.pid}-${testDbId++}.db`);
```
Comment (kulyk) [inline frontend/test/__runner__/backend.js:12]: Does it actually generate a DB?

Comment (nemanjaglumac) [reply frontend/test/__runner__/backend.js:12]: More precisely, it generates a string/path that is then passed to the `MB_DB_FILE` and is created during a `spawn` process. What would be better name for this? I'm blanking out.

File: frontend/test/__runner__/backend.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/__runner__/backend.js:62]: Would `connectionDefaults` be a better name?
Any suggestions (and alternative approaches) are more than welcome :)

File: frontend/test/__runner__/backend.js (hunk 2 above)
Comment (kulyk) [reply frontend/test/__runner__/backend.js:12]: WDYT about something like `generateTempDbFilePath` or `getTempDbPath`? Does it sound / represents the idea better?

**Review by gusaiani — APPROVED** [review]
Better 👍

**Review by kulyk — APPROVED** [review]

**Review by npfitz — APPROVED** [review]
A bit more readable!

**Pull request merged**